xbmc/settings/windows/GUIWindowSettingsScreenCalibration.cpp

```diff
@@ -505,11 +502,43 @@ bool CGUIWindowSettingsScreenCalibration::UpdateFromControl(int iControl)
   SET_CONTROL_LABEL(CONTROL_LABEL_RES, resInfo);
 
   // Detect overscan changes
-  bool overscanChanged = info.Overscan.bottom != infoPrev.Overscan.bottom ||
-                         info.Overscan.top != infoPrev.Overscan.top ||
-                         info.Overscan.left != infoPrev.Overscan.left ||
-                         info.Overscan.right != infoPrev.Overscan.right;
-  return overscanChanged;
+  bool isOverscanChanged = info.Overscan != infoPrev.Overscan;
+
+  // Adjust subtitle bar position due to overscan changes
+  if (isOverscanChanged)
+  {
+    CGUIMoverControl* pControl = dynamic_cast<CGUIMoverControl*>(GetControl(CONTROL_SUBTITLES));
+
+    // Keep the subtitle bar within the overscan boundary
+    if (info.Overscan.bottom + m_subtitleVerticalMargin < info.iSubtitles)
+    {
+      info.iSubtitles = info.Overscan.bottom + m_subtitleVerticalMargin;
+
+      // We want the text to be at the base of the bar,
+      // then we shift the position to include the vertical margin
+      pControl->SetPosition((info.iWidth - pControl->GetWidth()) * 0.5f,
+                            info.iSubtitles - pControl->GetHeight() + m_subtitlesHalfSpace -
+                                m_subtitleVerticalMargin);
+      pControl->SetLocation(0, info.iSubtitles + m_subtitlesHalfSpace - m_subtitleVerticalMargin,
+                            false);
+    }
+
+    // Recalculate limits based on overscan values
+    auto& size = m_controlsSize[CONTROL_SUBTITLES];
+    float scaledHeight = size.first / DEFAULT_GUI_HEIGHT * info.iHeight;
```

Comment:
This is more style preference, but I prefer `const float` for intermediate variables that aren't expected to change. Newer language like rust embody this philosophy by default-const.